pyvrp/cpp/search/TwoOpt.cpp

```diff
@@ -51,16 +51,40 @@ Cost TwoOpt::evalBetweenRoutes(Route::Node *U,
                                Route::Node *V,
                                CostEvaluator const &costEvaluator) const
 {
-    assert(U->route() && V->route());
+    if (n(U)->isDepot() && n(V)->isDepot())  // is a no-op
```

Comment:
This checks whether the route segments after U and V are empty. I don't think that'll happen all that often, so I have removed the check.

The old asserts were useless here, because the route members are already dereferenced before we get to this method (so they must have been something other than `nullptr`). I have moved those checks up to `TwoOpt::evaluate`.